### PR TITLE
Make checksumming the default behavior

### LINKED ISSUE #129: Make checksumming of downloaded products the default behavior
It is a good practice to verify the integrity of downloaded products and it makes sense to do it by default.
So, let's set `checksum` to `True` in `download()` and `download_all()` and change `--md5` to `--no-md5` in the CLI.
This is a backwards-incompatible change, so we probably want to do implement it in a release that bundles other such changes.

### PR DESCRIPTION
**Pull request by gkeller2**
Fixes #129 

As you can see, I had to modify the tests to remove the uses of the flag `--md5`, which is removed in this code change.

Also updated the README to remove the flag from the description of the CLI tool.

### CODE REVIEW

**Comment by Fernerkundung** [comment]
Thank you for taking time to contribute to this issue @gkeller2 :+1:

Before we merge this I'd like to ask you for some minor changes:

- please also change the default of the [`download`](https://github.com/sentinelsat/sentinelsat/blob/35666648385c07c8e429b1c06add826d3bd8cfe9/sentinelsat/sentinel.py#L387) function as well as the docstring
- remove the MD5 option in the [docs](https://github.com/sentinelsat/sentinelsat/blob/master/docs/cli.rst), just like you did in the Readme
- write the change into the [Changelog](https://github.com/sentinelsat/sentinelsat/blob/master/CHANGELOG.rst)
- and finally: add yourself to the list of contributors at the end of the Readme 😉

**Comment by gkeller2** [comment]
I'm not sure if I updated the Changelog correctly. Please, take a look.

File: sentinelsat/scripts/cli.py [hunk 1]
```diff
@@ -144,12 +141,11 @@ def cli(user, password, geometry, start, end, uuid, name, download, md5, sentine
             outfile.write(gj.dumps(footprints_geojson))
 
     if download is True:
-        product_infos, failed_downloads = api.download_all(products, path, checksum=md5)
-        if md5 is True:
-            if len(failed_downloads) > 0:
-                with open(os.path.join(path, "corrupt_scenes.txt"), "w") as outfile:
-                    for failed_id in failed_downloads:
-                        outfile.write("%s : %s\n" % (failed_id, products[failed_id]['title']))
+        product_infos, failed_downloads = api.download_all(products, path)
```
Comment (j08lue) [inline sentinelsat/scripts/cli.py]: Maybe keep the `checksum=True` here, to make the default behavior visible.

File: sentinelsat/sentinel.py [hunk 2]
```diff
@@ -401,7 +401,7 @@ def download(self, id, directory_path='.', checksum=False):
         checksum : bool, optional
             If True, verify the downloaded file's integrity by checking its MD5 checksum.
             Throws InvalidChecksumError if the checksum does not match.
-            Defaults to False.
+            Defaults to True.
```
Comment (j08lue) [inline sentinelsat/sentinel.py]: Why do we write this at all? The default is in the function declaration in this case (would be different if the default was `None`).

Comment (gkeller2) [reply sentinelsat/sentinel.py]: You are correct. I agree that there's little benefit to having a line in the method's doc specifying what the default value is, when it's defined in the method's signature. However, I did not write the original documentation -- I merely updated it.

File: sentinelsat/scripts/cli.py (hunk 1 above)
Comment (gkeller2) [reply sentinelsat/scripts/cli.py]: I find that unnecessary -- specifying a keyword parameter giving it its default value.

Comment (j08lue) [reply sentinelsat/scripts/cli.py]: Yes, you can see it that way, too.

File: sentinelsat/sentinel.py (hunk 2 above)
Comment (Fernerkundung) [reply sentinelsat/sentinel.py]: Documenting the default is optional according to the [numpy docstring standard](https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt#id5) that we use.

> Optional keyword parameters have default values, which are displayed as part of the function signature. They can also be detailed in the description:

We can remove it. In the case of this PR I'd say updating is the preferred choice. If we want to remove the documented defaults we should do it consistently throughout the codebase.

**Comment by Fernerkundung** [comment]
Thank you for your contribution @gkeller2! I will merge this PR soon and create a release.

**Pull request merged**